src/analysis/normed_space/enorm.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2020 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Yury G. Kudryashov
+-/
+import analysis.normed_space.basic
+
+/-!
+# Extended norm
+
+In this file we define structure `enorm k V` representing an extended norm (i.e., a norm that can
+take the value `∞`) on a vector space `V` over a normed field `k`. We do not use `class` for
+an `enorm` because the same space can have more than one extended norm. For example, the space of
+measurable functions `f : α → ℝ` has a family of `L_p` extended norms.
+
+We prove some basic inequalities, then define
+
+* `emetric_space` structure on `V` corresponding to `e : enorm k V`;
+* the subspace of vectors with finite norm;
+* a `normed_space` structure on this space.
+
+The last definition is an instance because the type involves `e`.
+
+## Implementation notes
+
+We do not define extended normed groups. They can be added to the chain once someone will need them.
+
+## Tags
+
+normed space, extended norm
+-/
+
+open_locale classical
+
+/-- Extended norm on a vector space. -/
+structure enorm (k : Type*) (V : Type*) [normed_field k] [add_comm_group V] [vector_space k V] :=
```

Comment:
Could you use \bbk for the field, here and elsewhere? Also, in the docstring, comment that `map_smul_le'` implies equality, but it stated like this because it is more convenient to prove sometimes.